machine_learning/DegradationModelBlackbox.py

```diff
@@ -512,113 +547,249 @@ def __init__(
         self.width = width
         self.n_channels = n_channels
 
+    def call(self, params, training = False):
+        """
+        Call function for the Model during training or evaluation.
+
+        Examples:
+
+            training:
+                ```python
+                train_results = degradation_model(
+                    (
+                        tf.expand_dims(cycle, axis = 1),
+                        tf.expand_dims(constant_current, axis = 1),
+                        tf.expand_dims(end_current_prev, axis = 1),
+                        tf.expand_dims(end_voltage_prev, axis = 1),
+                        tf.expand_dims(end_voltage, axis = 1),
+                        cell_indices,
+                        cc_voltage,
+                        cv_current,
+                        svit_grid,
+                        count_matrix,
+                    ),
+                    training = True,
+                )
+                ```
+
+            evaluation:
+                ```python
```

Comment:
is this still accurate? I think this is not still accurate. 
In my most recent version of this, we don't use len(voltages) because voltages is a tensor